itest/mint_batch_stress_test.go

```diff
@@ -30,29 +31,109 @@ var (
+	timeout := defaultWaitTimeout
+
+	// If we create a second tapd instance and sync the universe state,
+	// the synced tree should match the source tree.
+	bob := setupTapdHarness(
+		t.t, t, t.lndHarness.Bob, nil,
+	)
+	defer func() {
+		require.NoError(t.t, bob.stop(!*noDelete))
+	}()
+
+	mintBatches := func(reqs []*mintrpc.MintAssetRequest) []*taprpc.Asset {
+		return mintAssetsConfirmBatch(
+			t, t.tapd, reqs, withMintingTimeout(timeout),
+		)
+	}
+
+	getImageMetadataBytes := func() []byte {
+		return GetImageMetadataBytes(t.t, ImageMetadataFileName)
+	}
+
+	mintBatchStressTest(
+		t.t, t.tapd, bob, t.tapd.rpcHost(), 100, mintBatches,
+		getImageMetadataBytes, timeout,
+	)
 }
 
 func testMintBatch1kStressTest(t *harnessTest) {
-	mintBatchStressTest(t, 1000, defaultWaitTimeout*10)
+	timeout := defaultWaitTimeout * 10
+
+	// If we create a second tapd instance and sync the universe state,
+	// the synced tree should match the source tree.
+	bob := setupTapdHarness(
+		t.t, t, t.lndHarness.Bob, nil,
+	)
+	defer func() {
+		require.NoError(t.t, bob.stop(!*noDelete))
+	}()
+
+	mintBatches := func(reqs []*mintrpc.MintAssetRequest) []*taprpc.Asset {
+		return mintAssetsConfirmBatch(
+			t, t.tapd, reqs, withMintingTimeout(timeout),
+		)
+	}
+
+	getImageMetadataBytes := func() []byte {
+		return GetImageMetadataBytes(t.t, ImageMetadataFileName)
+	}
+
+	mintBatchStressTest(
+		t.t, t.tapd, bob, t.tapd.rpcHost(), 1000, mintBatches,
+		getImageMetadataBytes, timeout,
+	)
 }
 
 func testMintBatch10kStressTest(t *harnessTest) {
-	mintBatchStressTest(t, 10000, defaultWaitTimeout*100)
-}
+	timeout := defaultWaitTimeout * 100
 
-func mintBatchStressTest(t *harnessTest, batchSize int,
-	minterTimeout time.Duration) {
+	// If we create a second tapd instance and sync the universe state,
+	// the synced tree should match the source tree.
+	bob := setupTapdHarness(
+		t.t, t, t.lndHarness.Bob, nil,
+	)
+	defer func() {
+		require.NoError(t.t, bob.stop(!*noDelete))
+	}()
+
+	mintBatches := func(reqs []*mintrpc.MintAssetRequest) []*taprpc.Asset {
+		return mintAssetsConfirmBatch(
+			t, t.tapd, reqs, withMintingTimeout(timeout),
+		)
+	}
+
+	getImageMetadataBytes := func() []byte {
+		return GetImageMetadataBytes(t.t, ImageMetadataFileName)
+	}
+
+	mintBatchStressTest(
+		t.t, t.tapd, bob, t.tapd.rpcHost(), 10000, mintBatches,
+		getImageMetadataBytes, timeout,
+	)
+}
 
+// GetImageMetadataBytes returns the image metadata bytes from the given file.
+func GetImageMetadataBytes(t *testing.T, fileName string) []byte {
 	// Read base metadata.
-	imageMetadataHex, err := os.ReadFile(ImageMetadataFileName)
-	require.NoError(t.t, err)
+	imageMetadataHex, err := os.ReadFile(fileName)
+	require.NoError(t, err)
 
 	imageMetadataBytes, err := hex.DecodeString(
 		strings.Trim(string(imageMetadataHex), "\n"),
 	)
-	require.NoError(t.t, err)
+	require.NoError(t, err)
+
+	return imageMetadataBytes
+}
 
+func mintBatchStressTest(
```

Comment:
Is there some plan to make this public later on, and then call it from the new test binary? Or does it need further adjustments.

Alternatively, I'm misunderstanding which parts of the existing itests are going to get re-used, and this would get 'reimplemented' in the standalone binary.

Reply:
This test is going to be moved to the `perf_test` packages. I think no need to keep in the itest package but not execute it?

Reply:
makes sense - if it's in `perf_test` I think I'd want the test to still have a Makefile target somehow, or be able to run it locally as I can with the optional tests currently.